eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php

```diff
@@ -191,8 +174,95 @@ public function applyJoin(SelectQuery $query, SortClause $sortClause, $number)
                         $this->dbHandler->quoteColumn('version', $table),
                         $this->dbHandler->quoteColumn('current_version', 'ezcontentobject')
                     ),
-                    $languageExpression
+                    $this->getFieldCondition($query, $languageSettings, $table)
                 )
             );
     }
+
+    /**
+     * Returns a field language join condition for the given $languageSettings.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param array $languageSettings
+     * @param string $fieldTableName
+     *
+     * @return string
+     */
+    protected function getFieldCondition(SelectQuery $query, array $languageSettings, $fieldTableName)
```

Comment:
BTW, as far as I can see that implementation will need to use [CASE operator](https://dev.mysql.com/doc/refman/5.0/en/control-flow-functions.html#operator_case) to select sort columns, which is not abstracted and is in fact vendor-specific, but it should be possible.